File: ipman.shared/Entity/Department.cs
using ipman.shared.Entity.Join;
using System;
using System.Collections.Generic;
using System.Text;

namespace ipman.shared.Entity
{
    public class Department: EntityBase
    {
        public string DepartmentName { get; set; }
        public DateTime CreatedUTC { get; set; }
        public DateTime LastUpdatedUTC { get; set; }
        public bool IsActive { get; set; }
        public Guid SiteAccountID { get; set; }
        public SiteAccount SiteAccount { get; set; }
        public ICollection<SiteAccountUserAccountDepartment> SiteAccountUserAccountDepartments { get; set; }
    }
}

File: ipman/ClientApp/entity/Department.ts
import { EntityBase } from "./EntityBase";
import SiteAccount from "./SiteAccount";
import SiteAccountUserAccountDepartment from "./join/SiteAccountUserAccountDepartment";

export default interface Department extends EntityBase {
    DepartmentName: string;
    CreatedUTC: Date | string;
    LastUpdatedUTC: Date | string;
    IsActive: boolean;
    SiteAccount: SiteAccount;
    SiteAccountUserAccountDepartment: SiteAccountUserAccountDepartment[];
}